test/e2e/categories_test.go

```diff
@@ -174,4 +183,92 @@ var _ = Describe("Nutanix categories", Label("capx-feature-test", "categories"),
 
 		By("PASSED!")
 	})
+
+	It("Create and delete 2 clusters with same name with default cluster categories and should succeed", Label("same-name-two-cluster-test"), func() {
+		Expect(namespace).NotTo(BeNil())
+		flavor := clusterctl.DefaultFlavor
+		expectedClusterNameCategoryKey := infrav1.DefaultCAPICategoryKeyForName
+		By("Creating a workload cluster 1", func() {
+			testHelper.deployClusterAndWait(
+				deployClusterParams{
+					clusterName:           clusterName,
+					namespace:             namespace,
+					flavor:                flavor,
+					clusterctlConfigPath:  clusterctlConfigPath,
+					artifactFolder:        artifactFolder,
+					bootstrapClusterProxy: bootstrapClusterProxy,
+				}, clusterResources)
+		})
+
+		By("Checking cluster category condition is true", func() {
+			testHelper.verifyConditionOnNutanixCluster(verifyConditionParams{
+				clusterName:           clusterName,
+				namespace:             namespace,
+				bootstrapClusterProxy: bootstrapClusterProxy,
+				expectedCondition: clusterv1.Condition{
+					Type:   infrav1.ClusterCategoryCreatedCondition,
+					Status: corev1.ConditionTrue,
+				},
+			})
+		})
+
+		By("Checking if a category was created", func() {
+			testHelper.verifyCategoryExists(ctx, expectedClusterNameCategoryKey, clusterName)
+		})
+
+		By("Checking if there are VMs assigned to this category", func() {
+			expectedCategories := map[string]string{
+				expectedClusterNameCategoryKey: clusterName,
+			}
+			testHelper.verifyCategoriesNutanixMachines(ctx, clusterName, namespace.Name, expectedCategories)
+		})
+
+		By("Setting different Control plane endpoint IP for 2nd cluster", func() {
+			cp2EndpointIP := testHelper.getVariableFromE2eConfig("CONTROL_PLANE_ENDPOINT_IP_2")
```

Comment:
we set `CONTROL_PLANE_ENDPOINT_IP_WORKLOAD_CLUSTER` in CI as a second Control Plane endpoint IP if you need one.

Reply:
ok good to know. will use it then. currently updating it in the e2eConfig is not working and 2nd cluster gets same ip as first somehow. debugging

Reply:
had to do os.setenv to reflect the new cp ip